third_party/amd/lib/TritonAMDGPUTransforms/AccelerateAMDMatmul.cpp

```diff
@@ -591,10 +581,14 @@ class BlockedToMFMA : public OpRewritePattern<tt::DotOp> {
     SmallVector<unsigned, 2> tilesPerWarp{1, 1};
     if (mDim == nDim && (mDim == 32 || mDim == 16)) {
       SmallVector<unsigned, 2> tilesA{1, 1}, tilesB{1, 1};
-      int vecA = deduceTilesPerWarp(cast<TensorValue>(a), 0, mDim, warpsPerTile,
-                                    /*fromDecomposedScaledDot=*/true, &tilesA);
-      int vecB = deduceTilesPerWarp(cast<TensorValue>(b), 1, mDim, warpsPerTile,
-                                    /*fromDecomposedScaledDot=*/true, &tilesB);
+      Value scaleA = findScaleAsDecompositionSource(a);
+      Value scaleB = findScaleAsDecompositionSource(b);
+      int vecA = scaleA ? deduceTilesPerWarp(cast<TensorValue>(scaleA), 0, mDim,
+                                             warpsPerTile, &tilesA)
+                        : 1;
+      int vecB = scaleB ? deduceTilesPerWarp(cast<TensorValue>(scaleB), 1, mDim,
+                                             warpsPerTile, &tilesB)
+                        : 1;
```

Comment:
It's fine to just do `int vecA = deduceTilesPerWarp(..)` given it will return 1 anyway inside?